src/python_testing/TC_CLCTRL_4_3.py

```diff
@@ -0,0 +1,183 @@
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_CLCTRL_4_3(self) -> str:
+        return "[TC-CLCTRL-4.3] MoveTo Command Field Sanity Check with Server as DUT"
+
+    def steps_TC_CLCTRL_4_3(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Send MoveTo command with no fields"),
+            TestStep("3a", "Send MoveTo command with fields where corresponding feature is not supported - !Latch"),
+            TestStep("3b", "Send MoveTo command with fields where corresponding feature is not supported - !Speed"),
+            TestStep("3c", "Send MoveTo command with fields where corresponding feature is not supported - !Position"),
+            TestStep("4a", "Send MoveTo command with invalid data in fields - invalid Position"),
+            TestStep("4b", "Send MoveTo command with invalid data in fields - invalid Speed"),
+            TestStep("4c", "Send MoveTo command with invalid data in fields - invalid Position and valid Speed"),
+            TestStep("4d", "Send MoveTo command with invalid data in fields - valid Position and invalid Speed"),
+        ]
+        return steps
+
+    def pics_TC_CLCTRL_4_3(self) -> list[str]:
+        pics = [
+            "CLCTRL.S",
+            "CLCTRL.C.C01.Rsp(MoveTo)",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_CLCTRL_4_3(self):
+        
+        endpoint = self.get_endpoint(default=1)
+        
+        # Read the feature map
+        feature_map = await self.read_clctrl_attribute_expect_success(endpoint=endpoint, attribute=Clusters.ClosureControl.Attributes.FeatureMap)
+        logging.info(f"FeatureMap: {feature_map}")
+
+        is_ps_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kPosition
+        is_lt_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kLatch
+        is_sp_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kSpeed
+        
+        # STEP 1: Commission DUT to TH
+        self.step(1)
+        cluster = Clusters.Objects.ClosureControl
+        
+        # STEP 2: Send MoveTo command with no fields
+        self.step(2)
+        try:
+            response = await self.send_single_cmd(endpoint=endpoint, cluster=cluster, command=cluster.Commands.MoveTo({}))
+            logging.error("MoveTo command with no fields should have failed but succeeded")
+            asserts.assert_true(False, "MoveTo command with no fields should have failed but succeeded")
+        except Exception as e:
+            logging.info(f"Expected exception caught: {e}")
+            # Verify that we got an Invalid Command error
+            asserts.assert_true("INVALID_COMMAND" in str(e), 
+                                f"Expected INVALID_COMMAND error but got: {e}")
```

Comment:
if we catch an InteractionModel error we can validated the status matches instead of checking for the string.